spark/src/main/kotlin/com/adevinta/spark/components/tags/Tag.kt

```diff
@@ -0,0 +1,179 @@
+import androidx.compose.foundation.layout.size
+import androidx.compose.material3.LocalContentColor
+import androidx.compose.material3.ProvideTextStyle
+import androidx.compose.material3.Surface
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.CompositionLocalProvider
+import androidx.compose.runtime.Immutable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.adevinta.spark.InternalSparkApi
+import com.adevinta.spark.PreviewTheme
+import com.adevinta.spark.SparkTheme
+import com.adevinta.spark.components.icons.Icon
+import com.adevinta.spark.components.tags.TagDefaults.LeadingIconSize
+import com.adevinta.spark.components.tags.TagDefaults.MinHeight
+import com.adevinta.spark.icons.SparkIcon
+import com.adevinta.spark.tokens.contentColorFor
+import com.adevinta.spark.tools.modifiers.SlotArea
+import com.adevinta.spark.tools.modifiers.sparkUsageOverlay
+
+@InternalSparkApi
+@Composable
+internal fun SparkTag(
+    colors: TagColors,
+    modifier: Modifier = Modifier,
+    border: BorderStroke? = null,
+    leadingIcon: SparkIcon? = null,
+    tint: Color? = null,
+    content: @Composable RowScope.() -> Unit,
+) {
+    Surface(
+        modifier = modifier.sparkUsageOverlay(),
+        shape = SparkTheme.shapes.extraSmall,
+        color = colors.backgroundColor,
+        contentColor = colors.contentColor.copy(1.0f),
+        border = border,
+    ) {
+        ProvideTextStyle(
+            value = SparkTheme.typography.smallImportant,
+        ) {
+            Row(
+                Modifier
+                    .defaultMinSize(minHeight = MinHeight)
+                    .padding(vertical = VerticalPadding, horizontal = HorizontalPadding),
+                horizontalArrangement = Arrangement.spacedBy(LeadingIconEndSpacing, Alignment.Start),
+                verticalAlignment = Alignment.CenterVertically,
+            ) {
+                if (leadingIcon != null) {
+                    CompositionLocalProvider(
+                        LocalContentColor provides colors.contentColor,
+                    ) {
+                        Icon(
+                            sparkIcon = leadingIcon,
+                            modifier = Modifier.size(LeadingIconSize),
+                            contentDescription = null, // The tag is associated with a mandatory label so it's okay
+                            tint = tint ?: LocalContentColor.current,
+                        )
+                    }
+                }
+                content()
+            }
+        }
+    }
+}
+
+@Immutable
+data class TagColors(
+    val backgroundColor: Color,
+    val contentColor: Color,
+)
+
+object TagDefaults {
+    internal val MinHeight = 20.dp
+
+    /**
+     * The outlined chip's border size
```

Comment:
The wording "chip" is used in multiple files. I think we should replace it.

Reply:
Done in [51e9ff7](https://github.com/adevinta/spark-android/pull/155/commits/51e9ff78bdb5c3d4e31bcc5219472cb183c5cbea)